File: frontend/src/types/news.ts
export type Sentiment = 'positive' | 'negative' | 'neutral';

export interface BaseNewsItem {
  id: string;
  title: string;
  source: string;
  time: string;
  imageUrl: string;
  author?: string;
  sentiment: Sentiment;
}

export interface Story extends BaseNewsItem {}
export interface Pick extends BaseNewsItem {}



export type articleType = {
  url: string;
  read_more_link: string;
  language: string;
  title: string;
  top_image: string;
  meta_img: string;
  images: string[];
  movies: string[];
  keywords: string[];
  meta_keywords: string[];
  tags: string[] | null;
  authors: string[];
  publish_date: string | null;
  summary: string;
  meta_description: string;
  meta_lang: string;
  meta_favicon: string;
  source:string;
  date:string;
  meta_site_name: string;
  canonical_link: string;
  text: string;
  id: string;
};

export type fetchAllArticlesAPIResponse = articleType[];

File: backend/app/dto/news.py
from datetime import datetime
from typing import List, Optional
from pydantic import BaseModel


class News(BaseModel):
    id: str
    url: str
    read_more_link: Optional[str]
    language: str
    title: str
    top_image: Optional[str]
    meta_img: Optional[str]
    images: List[str] = []
    movies: List[str] = []
    keywords: List[str] = []
    meta_keywords: List[str] = []
    tags: List[str] = []
    authors: List[str] = []
    publish_date: Optional[datetime]
    summary: str
    meta_description: Optional[str]
    meta_lang: Optional[str]
    meta_favicon: Optional[str]
    meta_site_name: Optional[str]
    canonical_link: str
    text: str
    country: str
    decoded_url: str
    google_uri: str
    extracted_keywords: List[str]
    sentiment: str
    sentiment_impactful_texts: List[str]
